pyne/alara.py

```diff
@@ -287,31 +289,51 @@ def record_to_geom(mesh, cell_fracs, cell_mats, geom_file, matlib_file,
     mixture = '' # mixture blocks
 
     unique_mixtures = []
-    for i, mat, ve in mesh:
-        volume += '    {0: 1.6E}    zone_{1}\n'.format(mesh.elem_volume(ve), i)
-
-        ve_mixture = {}
-        for row in cell_fracs[cell_fracs['idx'] == i]:
-            cell_mat = cell_mats[row['cell']]
-            name = cell_mat.metadata['name']
-            if _is_void(name):
-                name = 'mat_void'
-            if name not in ve_mixture.keys():
-                ve_mixture[name] = np.round(row['vol_frac'], sig_figs)
-            else:
-                ve_mixture[name] += np.round(row['vol_frac'], sig_figs)
-
-        if ve_mixture not in unique_mixtures:
-            unique_mixtures.append(ve_mixture)
-            mixture += 'mixture mix_{0}\n'.format(
-                                           unique_mixtures.index(ve_mixture))
-            for key, value in ve_mixture.items():
-                mixture += '    material {0} 1 {1}\n'.format(key, value)
-
-            mixture += 'end\n\n'
+    if not sub_voxel:
+        for i, mat, ve in mesh:
+            volume += '    {0: 1.6E}    zone_{1}\n'.format(
+                mesh.elem_volume(ve), i)
+
+            ve_mixture = {}
+            for row in cell_fracs[cell_fracs['idx'] == i]:
+                cell_mat = cell_mats[row['cell']]
+                name = cell_mat.metadata['name']
+                if _is_void(name):
+                    name = 'mat_void'
+                if name not in ve_mixture.keys():
+                    ve_mixture[name] = np.round(row['vol_frac'], sig_figs)
+                else:
+                    ve_mixture[name] += np.round(row['vol_frac'], sig_figs)
+
+            if ve_mixture not in unique_mixtures:
+                unique_mixtures.append(ve_mixture)
+                mixture += 'mixture mix_{0}\n'.format(
+                                            unique_mixtures.index(ve_mixture))
+                for key, value in ve_mixture.items():
+                    mixture += '    material {0} 1 {1}\n'.format(key, value)
+
+                mixture += 'end\n\n'
+
+            mat_loading += '    zone_{0}    mix_{1}\n'.format(i,
+                            unique_mixtures.index(ve_mixture))
+    else:
+        sve_count = 0
+        for row in cell_fracs:
+            for i, mat, ve in mesh:
```

Comment:
I just noticed the same in #936

Reply:
Maybe this is related to how we ensure consistent ordering?  but this loop ordering is going to be extremely!!! Inefficient for large mesh:  For a mesh with N voxels, `cell_fracs` will have O(N) rows and mesh will have N entries, so it will be N^2